controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *odfInfoConfig) getOdfInfoKeyName(storageCluster *ocsv1.StorageCluster) string {
+	return fmt.Sprintf("%s/%s.%s", storageCluster.Namespace, storageCluster.Name, odfInfoKeySuffix)
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	odfInfoConfigMap := &corev1.ConfigMap{}
+	odfInfoConfigMap.Name = odfInfoConfigMapName
+	odfInfoConfigMap.Namespace = operatorNamespace
+	if err = r.Client.Get(r.ctx, client.ObjectKeyFromObject(odfInfoConfigMap), odfInfoConfigMap); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	if len(odfInfoConfigMap.Data) > 1 {
+		odfInfoKeyName := obj.getOdfInfoKeyName(storageCluster)
+		delete(odfInfoConfigMap.Data, odfInfoKeyName)
+		if err = r.Client.Update(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to update odf-info config map with deleted key.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name), "Key", odfInfoKeyName)
+			return reconcile.Result{}, fmt.Errorf("failed to delete key %v in odf-info %v: %v", odfInfoKeyName, odfInfoConfigMap.Name, err)
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info config map.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := odfDeploymentTypeExternal
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = odfDeploymentTypeInternal
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+
+	connectedClients, err := getConnectedClients(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	data := OdfInfoData{
+		Version:           odfVersion,
+		DeploymentType:    odfDeploymentType,
+		StorageSystemName: storageSystemName,
+		Clients:           connectedClients,
+		StorageCluster: infoStorageCluster{
+			NamespacedName:          client.ObjectKeyFromObject(storageCluster),
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
```

Comment:
Hey, I was not able the find the field isDROptimized in the configMap, I think it is required for [bluestore-rdr](https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/clusterclaims.go#L105-L114), it might break the already existing flow if we don't add this field.

Reply:
See [comment](https://issues.redhat.com/browse/RHSTOR-5117?focusedId=24677010&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24677010)